pyomo/contrib/alternative_solutions/solution.py

```diff
@@ -9,150 +9,285 @@
 #  This software is distributed under the 3-clause BSD License.
 #  ___________________________________________________________________________
 
+import sys
+import heapq
+import collections
+import dataclasses
 import json
+import functools
+
 import pyomo.environ as pyo
-from pyomo.common.collections import ComponentMap, ComponentSet
-from pyomo.contrib.alternative_solutions import aos_utils
 
+from .aos_utils import MyMunch, _to_dict
 
-class Solution:
+nan = float("nan")
+
+
+def _custom_dict_factory(data):
+    return {k: _to_dict(v) for k, v in data}
+
+
+if sys.version_info >= (3, 10):
+    dataclass_kwargs = dict(kw_only=True)
+else:
+    dataclass_kwargs = dict()
+
+
+@dataclasses.dataclass(**dataclass_kwargs)
+class VariableInfo:
     """
-    A class to store solutions from a Pyomo model.
+    Represents a variable in a solution.
 
     Attributes
     ----------
-    variables : ComponentMap
-        A map between Pyomo variables and their values for a solution.
-    fixed_vars : ComponentSet
-        The set of Pyomo variables that are fixed in a solution.
-    objective : ComponentMap
-        A map between Pyomo objectives and their values for a solution.
-
-    Methods
-    -------
-    pprint():
-        Prints a solution.
-    get_variable_name_values(self, ignore_fixed_vars=False):
-        Get a dictionary of variable name-variable value pairs.
-    get_fixed_variable_names(self):
-        Get a list of fixed-variable names.
-    get_objective_name_values(self):
-        Get a dictionary of objective name-objective value pairs.
+    value : float
+        The value of the variable.
+    fixed : bool
+        If True, then the variable was fixed during optimization.
+    name : str
+        The name of the variable.
+    index : int
+        The unique identifier for this variable.
+    discrete : bool
+        If True, then this is a discrete variable
```

Comment:
Would it be more general to just store the Var's domain, and then have a utility to check it it's discrete?